libfreerdp/core/connection.c

```diff
@@ -1264,6 +1264,21 @@ state_run_t rdp_client_connect_demand_active(rdpRdp* rdp, wStream* s)
 	if (freerdp_shall_disconnect_context(rdp->context))
 		return STATE_RUN_QUIT_SESSION;
 
+	if (rdp->mcs->messageChannelId && (channelId == rdp->mcs->messageChannelId))
```

Comment:
Looks ok for me, but don't we need to mutualize that, as I guess that we might get some inflight packets in other states, and also get some different packets ?

Reply:
@hardening good point. waiting for feedback, and if yes wrap this into a common function.

Reply:
ok, done, refactored this where possible.